controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -514,79 +536,70 @@ def test_create_snapshot_not_valid(self):
+        self._prepare_mocks_for_delete_snapshot()
         self.client_mock.get_volume.side_effect = NotFound("404")
         with self.assertRaises(array_errors.ObjectNotFoundError):
             self.array.delete_snapshot("fake_name")
 
     def _prepare_mocks_for_copy_to_existing_volume(self):
-        volume = Munch(
-            {"cap": "1073741824",
-             "id": "0001",
-             "name": "test_name",
-             "pool": "fake_pool",
-             "tp": "ese",
-             "flashcopy": ""
-             }
-        )
+        volume = self.volume_response
         self.client_mock.get_volumes_by_pool.side_effect = [[volume], [Munch(
             {"cap": "1073741824",
              "id": "0002",
-             "name": "snapshot_name",
+             "name": "source_name",
              "pool": "fake_pool",
              "tp": "ese",
              "flashcopy": []
              }
         )]]
         self.client_mock.get_volume.return_value = volume
         self.client_mock.get_flashcopies_by_volume.side_effect = \
-            [[], [self.flashcopy_from_volume_response], [self.flashcopy_from_volume_response]]
-        self.client_mock.get_flashcopies.return_value = self.flashcopy_from_volume_response
+            [[], [self.flashcopy_response], [self.flashcopy_response]]
+        self.client_mock.get_flashcopies.return_value = self.flashcopy_response
         self.client_mock.create_flashcopy.return_value = self.flashcopy_response
         return volume
 
-    def test_extend_volume(self):
+    def test_copy_to_existing_volume_with_extend_volume(self):
         volume = self._prepare_mocks_for_copy_to_existing_volume()
-        self.array.copy_to_existing_volume_from_source("test_name", "snapshot_name", 3, 2, "fake_pool")
+        self.array.copy_to_existing_volume_from_source("test_name", "source_name", 3, 2, "fake_pool")
         self.client_mock.extend_volume.assert_called_once_with(volume_id=volume.id,
                                                                new_size_in_bytes=3)
-
-    def test_extend_volume_not_found(self):
-        self._prepare_mocks_for_copy_to_existing_volume()
-        self.client_mock.extend_volume.side_effect = NotFound("404")
-        with self.assertRaises(array_errors.ObjectNotFoundError):
-            self.array.copy_to_existing_volume_from_source("test_name", "snapshot_name", 3, 2, "fake_pool")
-
-    def test_copy_to_existing_volume_flashcopy(self):
-        volume = self._prepare_mocks_for_copy_to_existing_volume()
-        self.array.copy_to_existing_volume_from_source("test_name", "snapshot_name", 3, 2, "fake_pool")
         self.client_mock.create_flashcopy.assert_called_once_with(
             source_volume_id="0002",
             target_volume_id=volume.id,
             options=[FLASHCOPY_PERSISTENT_OPTION,
                      FLASHCOPY_PERMIT_SPACE_EFFICIENT_TARGET_OPTION
                      ])
+
+    def test_clone_volume_not_found(self):
```

Comment:
since this test function doesn't assume the source type, and since we have a convention to begin with the name of the tested function, please avoid "clone"

Reply:
done